### ISSUE TITLE
Adding team-wide permissions to an org does not update team permissions

### ISSUE THREAD

**Issue by dsesami** [ansible/awx#2601]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - API
 - UI

##### SUMMARY
Clicking the + sign and adding team-wide permissions doesn't update the user permissions at all, or at least it does so inconsistently. 

![screenshot from 2018-11-05 16-12-20](https://user-images.githubusercontent.com/19942822/48027019-a4c90380-e115-11e8-8cfe-cf1db5089a40.png)

calls to /roles return a 204 no content, always, regardless of success or failure. 

##### ENVIRONMENT
* AWX version: 2.0.1-225-g895a567ed
* AWX install method: docker compose on fedora
* Ansible version:  2.7
* Operating System: fedora
* Web Browser: chrome

##### STEPS TO REPRODUCE

Create an org, put a team inside of it. Create a user, and put that user inside the team as well.

Go to Organizations from the left nav > click your org > go to the permissions tab and open the modal for adding permissions.

Click the teams tab for team-wide permissions, and attempt to add Read (or any) permission to the whole team.

##### EXPECTED RESULTS
The user, inside of the team, is updated on the list with read permissions.

##### ACTUAL RESULTS
The user list of permissions is not updated.

**Comment by jakemcdermott** [comment]
@dsesami 

After taking a look at this issue, I ran into two situations where the permissions list wasn't updating in the manner you described:

1. It never updates for `admin` and `member` role assignment - this is due to the same underlying restriction we saw in [#2563](https://github.com/ansible/awx/issues/2563#issuecomment-435741153) for some types of organization `object_roles` when dealing with teams specifically. I've created a PR to remove those options from the ui here: https://github.com/ansible/awx/pull/2605

2. The team that is being assigned the roles must have at least one (non superuser!) user, otherwise there will be no visible change in the list. (Seems obvious now that I write it down but I hit this one first!)
